integration-tests/e2e/emaildomain.create.cy.js

```diff
@@ -0,0 +1,83 @@
+const MenuPage = require('../pages/menuPage')
+const EmailDomainListingPage = require('../pages/emailDomainListingPage')
+const EmailDomainCreationPage = require('../pages/emailDomainCreationPage')
+const AuthErrorPage = require('../pages/authErrorPage')
+
+context('CreateEmailDomain', () => {
+  before(() => {
+    cy.clearCookies()
+  })
+
+  beforeEach(() => {
+    cy.task('reset')
+  })
+
+  function navigateToCreateEmailDomainPage() {
+    cy.task('stubSignIn', { roles: [{ roleCode: 'MAINTAIN_EMAIL_DOMAINS' }] })
+    cy.signIn()
+    cy.task('stubAllEmailDomains', {})
+    MenuPage.verifyOnPage().viewEmailDomainListing()
+    const emailDomainListingPage = EmailDomainListingPage.verifyOnPage()
+    emailDomainListingPage.navigateToCreateEmailDomainPageButton().click()
+    return EmailDomainCreationPage.verifyOnPage()
+  }
+
+  it('Should disallow attempting to reach create-email-domain page directly via URL, if unauthorised', () => {
+    cy.task('stubSignIn', {
+      roles: [{ roleCode: 'NOT_MAINTAIN_EMAIL_DOMAINS' }],
+    })
+    cy.signIn()
+    MenuPage.verifyOnPage()
+    cy.get('[data-qa="view_email_domains_link"]').should('not.exist')
+    cy.visit('/create-email-domain', { failOnStatusCode: false })
+    AuthErrorPage.verifyOnPage()
+  })
+
+  it('Should create new email domain, after completing form and clicking the create email domain button', () => {
+    const emailDomainCreationPage = navigateToCreateEmailDomainPage()
+    emailDomainCreationPage.createEmailDomainPage('Domain1', 'Domain1Description')
+    cy.task('verifyGetAllEmailDomains').should((requests) => {
```

Comment:
I don't think this is still quite right. We are still only checking that the getAllDomainsList has been called - not what the test specifies. If you're doing the test using the verify - it should verifyAddEmailDomain method is called. Are you stubbing this method out?

Reply:
Please, have a look at the code and ping me for a hurdle. We can be having long chatty conversation here! Its easier.

Reply:
@karenmillermoj - Ping me on slack when you have a minute!